src/Features/Core/Portable/CodeLens/ReferenceLocationDescriptor.cs

```diff
@@ -9,14 +9,35 @@ namespace Microsoft.CodeAnalysis.CodeLens
     /// </summary>
     internal sealed class ReferenceLocationDescriptor
     {
-        public int LineNumber { get; }
-
-        public int ColumnNumber { get; }
-
         public Guid ProjectGuid { get; }
 
         public Guid DocumentGuid { get; }
 
+        /// <summary>
+        /// Document's file path
+        /// </summary>
+        public string FilePath { get; }
+
+        /// <summary>
+        /// Reference's span start based on the document content
+        /// </summary>
+        public int SpanStart { get; }
+
+        /// <summary>
+        /// Reference's span length based on the document content
+        /// </summary>
+        public int SpanLength { get; }
+
+        /// <summary>
+        /// Reference's line based on the document content
+        /// </summary>
+        public int LineNumber { get; }
+
+        /// <summary>
+        /// Reference's character based on the document content
+        /// </summary>
+        public int ColumnNumber { get; }
```

Comment:
i'm a bit confused by this one.  it says "character based", but is called 'Column'.  Also, the summary is a bit confusing.

Reply:
This should probably be moved to the review of #30865, as part of ensuring the refactoring there is complete/consistent.

Reply:
not cleaning up code lens part. I am leaving things as it is. also, this type is shared between VS and roslyn, changing here will break VS build.